### PR TITLE
feat: add isValidNumber, isNotValidNumber

### PR DESCRIPTION
**Pull request by Undistraction**
- adds `isValidNumber` which tests whether a value is a number that is not either `NaN`, `Infinity` or `-Infinity`.
- adds `isNotValidNumber` which is the complement of `isValidNumber`.
- adds `isInvalidNumber` which is aliased to `isNotValidNumber`.

Ref #235

Please note the following:

## Typescript

I haven't added the three new functions to `index.d.ts` as I have no familiarity with TypeScript and I would be doing so blindly.

## Docs

I have set the `@since` versions at X.X.X as I don't know what version you will include this feat in.

### npm test

I have one failing test which was already failing when I pulled the branch and is unrelated to my changes. If you want me to dig deeper please let me know and I'll see if I can work out why it's failing on my machine:

```
isPromise when value is an instance of native Promise should return true:

      AssertionError: 'false' === 'true'
      + expected - actual

      -false
      +true
```

### npm test:ramda

Initally failed because of unmet peer dependency of `ajv`. I upgraded `ajv` to @5 and all pass other than the test outlined above.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -53,6 +53,9 @@ export { default as isInteger } from './isInteger';
 export { default as isNotInteger } from './isNotInteger';
 export { default as isFloat } from './isFloat';
 export { default as isNotFloat } from './isNotFloat';
+export { default as isValidNumber } from './isValidNumber';
+export { default as isNotValidNumber } from './isNotValidNumber';
+export { default as isInvalidNumber } from './isNotValidNumber'; // alias of isNotValidNumber
```
Comment (char0n) [inline src/index.js]: IMHO this is somehow misleading naming for `isInvalidNumber`. The name indicates that the value checked by the predicate will be number, but not a valid one e.g. NaN, Infinity.

I would imagine isInvalidNumber to have the following signature

```javascript
const isInvalidNumber = both(isNumber, isNotValidNumber);
```

File: src/isNotValidNumber.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { complement } from 'ramda';
+import isValidNumber from './isValidNumber';
+
+/* eslint-disable max-len */
```
Comment (char0n) [inline src/isNotValidNumber.js]: Drop this. Your lines length should not go over 100 chars

File: src/isNotValidNumber.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * @category Type
+ * @sig * -> Boolean
+ * @param {*} val The value to test
+ * @return {Boolean}
+ * @example
+ *
+ * RA.isNotValidNumber(1); //=> false
+ * RA.isNotValidNumber(''); //=> true
+ */
+/* eslint-enable max-len */
```
Comment (char0n) [inline src/isNotValidNumber.js]: Drop this. Your lines length should not go over 100 chars

File: src/isNotValidNumber.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import isValidNumber from './isValidNumber';
+
+/* eslint-disable max-len */
+/**
+ * Checks if value is not a valid `Number` object or is `NaN`,
+ * `POSITIVE_INFINITY` or `NEGATIVE_INFINITY`.
+ *
+ * @func isNotValidNumber
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/X.X.X|vX.X.X}
```
Comment (char0n) [inline src/isNotValidNumber.js]: /2.2.0/v2.2.0

File: src/isNotValidNumber.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { complement } from 'ramda';
```
Comment (char0n) [inline src/isNotValidNumber.js]: divide absolute and relative imports by one empty line pls

File: src/isNotValidNumber.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import isValidNumber from './isValidNumber';
+
+/* eslint-disable max-len */
+/**
+ * Checks if value is not a valid `Number` object or is `NaN`,
```
Comment (char0n) [inline src/isNotValidNumber.js]: If we gonna drop the `Number` prexif, then let's use proper global object naming
```javascript
POSITIVE_INFINITY => Infinity
NEGATIVE_INFINITY => -Infinity

File: src/isNotValidNumber.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+ * @func isNotValidNumber
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/X.X.X|vX.X.X}
+ * @category Type
+ * @sig * -> Boolean
+ * @param {*} val The value to test
+ * @return {Boolean}
+ * @example
+ *
+ * RA.isNotValidNumber(1); //=> false
```
Comment (char0n) [inline src/isNotValidNumber.js]: I'd add examples also for NaN, Infinity and -Infinity

File: src/isNotValidNumber.js (hunk 6 above)
Comment (char0n) [inline src/isNotValidNumber.js]: IMHO this function description is efficient.

```
Checks if value is not a valid `Number`. A valid `Number` is a number that is not `NaN`, `Infinity` or `-Infinity`.

File: src/isNotValidNumber.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+/* eslint-disable max-len */
+/**
+ * Checks if value is not a valid `Number` object or is `NaN`,
+ * `POSITIVE_INFINITY` or `NEGATIVE_INFINITY`.
+ *
+ * @func isNotValidNumber
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/X.X.X|vX.X.X}
+ * @category Type
+ * @sig * -> Boolean
```
Comment (char0n) [inline src/isNotValidNumber.js]: Add 
```
@see {@link RA.isValidNumber|isValidNumber}

File: src/isValidNumber.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import { either } from 'ramda';
+
+import isFloat from './isFloat';
+import isInteger from './isInteger';
+
+/* eslint-disable max-len */
```
Comment (char0n) [inline src/isValidNumber.js]: Drop this. Your lines length should not go over 100 chars

File: src/isValidNumber.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * Checks if value is valid `Number` object and not `NaN`,
```
Comment (char0n) [inline src/isValidNumber.js]: Checks if value is a valid `Number`. A valid `Number` is a number that is not `NaN`, `Infinity` or `-Infinity`.

File: src/isValidNumber.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+ * `INFINITY` or `-INFINITY`.
+ *
+ * @func isValidNumber
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/X.X.X|vX.X.X}
```
Comment (char0n) [inline src/isValidNumber.js]: 2.2.0

File: src/isValidNumber.js [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+ * @category Type
+ * @sig * -> Boolean
```
Comment (char0n) [inline src/isValidNumber.js]: Add
```javascript
@see {@link RA.isNotValidNumber|isNotValidNumber}

File: test/isNotValidNumber.js [hunk 13]
```diff
@@ -0,0 +1,47 @@
+import * as RA from '../src/index';
+import MAX_SAFE_INTEGER from '../src/internal/polyfills/Number.MAX_SAFE_INTEGER';
+import MIN_SAFE_INTEGER from '../src/internal/polyfills/Number.MIN_SAFE_INTEGER';
+import eq from './shared/eq';
+import args from './shared/arguments';
+import Symbol from './shared/Symbol';
+
```
Comment (char0n) [inline test/isNotValidNumber.js]: insert empty line

File: test/isNotValidNumber.js [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+    if (Symbol !== 'undefined') {
+      eq(RA.isNotValidNumber(Symbol), true);
+    }
+
+    eq(RA.isNotValidNumber(null), true);
+    eq(RA.isNotValidNumber(undefined), true);
+  });
+});
+
+describe('isInvalidNumber', function() {
```
Comment (char0n) [inline test/isNotValidNumber.js]: Please drop this one. We'll not deal with `isInvalidNumber` in this PR.

File: test/isValidNumber.js [hunk 15]
```diff
@@ -0,0 +1,41 @@
+import * as RA from '../src/index';
+import MAX_SAFE_INTEGER from '../src/internal/polyfills/Number.MAX_SAFE_INTEGER';
+import MIN_SAFE_INTEGER from '../src/internal/polyfills/Number.MIN_SAFE_INTEGER';
+import eq from './shared/eq';
+import args from './shared/arguments';
+import Symbol from './shared/Symbol';
+
```
Comment (char0n) [inline test/isValidNumber.js]: insert another empty line

**Comment by char0n** [comment]
Regarding the `isPromise` not passing. I'am not able to reproduce. That is really really strange because these tests are running inside node.

**Comment by char0n** [comment]
Typescript signatures will looks like this:
```typescript
/**
* ... description
*/
isValidNumber(val: any): boolean;

/**
* ... description
*/
isNotValidNumber(val: any): boolean;

File: src/isValidNumber.js [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+ * @param {*} val The value to test
+ * @return {Boolean}
+ * @example
+ *
+ * RA.isValidNumber(1); //=> true
```
Comment (char0n) [inline src/isValidNumber.js]: I'd add examples also for NaN, Infinity and -Infinity

File: src/isValidNumber.js [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+ * RA.isValidNumber(''); //=> false
+ */
+/* eslint-enable max-len */
```
Comment (char0n) [inline src/isValidNumber.js]: Drop this. Your lines length should not go over 100 chars

File: src/isNotValidNumber.js (hunk 3 above)
Comment (Undistraction) [reply src/isNotValidNumber.js]: Not sure what you mean.  What should I drop?

File: src/isValidNumber.js (hunk 9 above)
Comment (Undistraction) [reply src/isValidNumber.js]: Not sure what you mean.  What should I drop?

File: src/isNotValidNumber.js (hunk 3 above)
Comment (char0n) [reply src/isNotValidNumber.js]:
```javascript
/* eslint-enable max-len */

File: src/isValidNumber.js (hunk 9 above)
Comment (char0n) [reply src/isValidNumber.js]:
```javascript
/* eslint-disable max-len */

File: src/isNotValidNumber.js (hunk 3 above)
Comment (Undistraction) [reply src/isNotValidNumber.js]: Ah. Sorry. Though you'd wrapped all your comment blocks in the enable/disable comments, but was just the one I cribbed from.

**Comment by Undistraction** [comment]
All changes made and I have removed `isInvalidNumber`.

**Pull request merged**